File: Exercism Codes/Python/all_your_base.py
def rebase(input_base, digits, output_base):
    if input_base < 2:
        raise ValueError("input base must be >= 2")
    if output_base < 2:
        raise ValueError("output base must be >= 2")

    result = []
    dec = 0
    n = len(digits)

    for idx, d in enumerate(digits):
        if d < 0 or d >= input_base:
            raise ValueError("all digits must satisfy 0 <= d < input base")

        dec += pow(input_base, n - idx - 1) * d

    while dec > 0:
        result.insert(0, dec % output_base)
        dec //= output_base

    return result if result else [0]

File: Exercism Codes/C#/AllYourBase.cs
public static class AllYourBase
{
    public static int[] Rebase(int inputBase, int[] inputDigits, int outputBase)
    {
        if (inputBase < 2)
        {
            throw new ArgumentException("Input base must be >= 2.");
        }
        if (outputBase < 2)
        {
            throw new ArgumentException("Output base must be >= 2.");
        }

        List<int> result = new List<int>();
        int dec = 0;
        int n = inputDigits.Length;

        for (int i = 0; i < n; i++)
        {
            if (inputDigits[i] < 0 || inputDigits[i] >= inputBase)
            {
                throw new ArgumentException("All digits must satisfy 0 <= d < input base.");
            }

            dec += (int)Math.Pow(inputBase, n - i - 1) * inputDigits[i];
        }

        while (dec > 0)
        {
            result.Insert(0, dec % outputBase);
            dec /= outputBase;
        }

        return result.Count == 0 ? [0] : result.ToArray();
    }
}